.github/workflows/automated-tests.yml

```
@@ -257,24 +257,54 @@ jobs:
+          TIMEOUT=1200  # 20 minutes
+          MAX_RETRIES=3
+          RETRY_COUNT=0
+          SUCCESS=0
+
+          while [[ $RETRY_COUNT -lt $MAX_RETRIES ]]; do
+            echo "Attempt $((RETRY_COUNT + 1)) of $MAX_RETRIES to install BBB..."
+
+            # Run the command with timeout and handle its exit code
+            # Capture both stdout and stderr
+            OUTPUT=$(timeout $TIMEOUT $COMMAND 2>&1) || COMMAND_EXIT_CODE=$?
+            
```

yamllint

[error] trailing spaces

(trailing-spaces)